src/java.net.http/share/classes/jdk/internal/net/http/Http2Connection.java

```diff
@@ -700,7 +733,9 @@ void shutdown(Throwable t) {
             closed = true;
         }
         if (Log.errors()) {
-            if (!(t instanceof EOFException) || isActive()) {
+            if (idleConnectionTimeoutEvent != null && idleConnectionTimeoutEvent.isFired()) {
```

Comment:
The reference for idleConnectionTimeoutEvent should be captured in the synchronized block above.

```
        IdleConnectionTimeoutEvent idleConnectionTimeoutEvent;
        synchronized (this) {
            if (closed == true) return;
            closed = true;
            idleConnectionTimeoutEvent = this.idleConnectionTimeoutEvent;
        }
        if (Log.errors()) {
            if (idleConnectionTimeoutEvent != null && idleConnectionTimeoutEvent.isFired()) { ... }
```

Reply:
And also, the check for Log.errors() could be ignored. Thinking that if the developer is configuring this property, the error should always be shown if the timeout occurs for clarity. An additional check for the correct exception type might be nice too in case the idleConnectionTimeoutEvent log is incorrectly logged when the event is not null _and_ has been fired. An unlikely case but worth catching perhaps.